core/src/main/java/br/com/brainweb/interview/core/features/hero/HeroRepository.java

```diff
@@ -16,9 +21,35 @@
         " (name, race, power_stats_id)" +
         " VALUES (:name, :race, :powerStatsId) RETURNING id";
 
+    private static final String FIND_HERO_BY_ID_QUERY = "SELECT * FROM hero "+
+        "INNER JOIN power_stats ON hero.power_stats_id = power_stats.id " +
+        "WHERE hero.id = :id";
+
+    private static final String FIND_HERO_BY_NAME_QUERY = "SELECT * FROM hero "+
+        "INNER JOIN power_stats ON hero.power_stats_id = power_stats.id " +
+        "WHERE hero.name LIKE :name";
+
+    private static final String UPDATE_HERO_QUERY = "UPDATE hero " +
+        "SET name = :name, race = :race, updated_at = now() " +
+        "WHERE id = :id RETURNING power_stats_id";
+
+    private static final String UPDATE_HERO_STATS_QUERY = "UPDATE power_stats " +
+        "SET strength = :strength, agility = :agility, " +
+        "dexterity = :dexterity, intelligence = :intelligence, updated_at = now() " +
+        "WHERE id = id";
+
+    private static final String DELETE_HERO_QUERY = "DELETE FROM hero " +
+        "WHERE hero.id = :id RETURNING id";
+
+    private static final String COMPARE_HEROES_QUERY = "SELECT power_stats.id, power_stats.strength, " +
+        "power_stats.agility, power_stats.dexterity, power_stats.intelligence FROM hero "+
+        "INNER JOIN power_stats ON hero.power_stats_id = power_stats.id " +
+        "WHERE hero.id = :id1 OR hero.id = :id2";
```

Comment:
Essa consulta vai retornar sempre o primeiro que encontrar, ou seja, nesse caso passando 2 ids nunca vai retornar os dois, acredito que para funcionar deveria ser um:
WHERE hero.id IN (:ids), faz sentido?

Reply:
Eu fiz testes aqui e a consulta retorna os dois. Se tivessem 3 dados com id1 e 2 com id2, retornaria as 5 linhas da tabela.
De qualquer maneira, sua sugestão é mais elegante tanto do ponto de vista de elegibilidade quanto de escalabilidade. Confesso que estava um pouco enferrujado aqui. hehe.